docs/apache-airflow/concepts/dynamic-task-mapping.rst

```diff
@@ -0,0 +1,256 @@
+        consumer.expand(arg=make_list())
+
+The make_list runs as a normal task and must return a list or dict (see `What data types can be expanded?`_), and then the consumer task will be called four times, once with each value in the return of make_list.
+
+Mapping with non-TaskFlow operators
+===================================
+
+It is possible to use partial and expand with classic style operators as well. Some arguments are not mappable, such as task_id, queue, pool, and most other arguments to BaseOperator and have to be passed to ``partial()``
+
+
+.. code-block:: python
+
+    BashOperator.partial(task_id="bash", do_xcom_push=False).expand(
+        bash_command=["echo 1", "echo 2"]
+    )
+
+Mapping over result of classic operators
+----------------------------------------
+
+If you want to map over the result of a classic operator you will need to create an XComArg object manually:
+
+.. code-block:: python
+
+    from airflow import XComArg
+
+    task = MyOperator(task_id="source")
+
+    downstream = MyOperator2.partial(task_id="consumer").expand(input=XComArg(task))
+
+Putting it all together
+=======================
+
+In this example you have a regular data delivery to an S3 bucket and want to apply the same processing to every file that arrives, no matter how many arrive each time.
+
+.. code-block:: python
+
+    from datetime import datetime
+
+    from airflow import DAG, XComArg
+    from airflow.decorators import task
+    from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+    from airflow.providers.amazon.aws.operators.s3 import S3ListOperator
+
+
+    with DAG(dag_id="mapped_s3", start_date=datetime(2020, 4, 7)) as dag:
+        files = S3ListOperator(
+            task_id="get_input",
+            bucket="example-bucket",
+            prefix='incoming/provider_a/{{ data_interval_start.strftime("%Y-%m-%d") }}',
+        )
+
+        @task
+        def count_lines(aws_conn_id, bucket, file):
+            hook = S3Hook(aws_conn_id=aws_conn_id)
+
+            return len(hook.read_key(file, bucket).splitlines())
+
+        @task
+        def total(lines):
+            return sum(lines)
+
+        counts = count_lines.partial(aws_conn_id="aws_default", bucket=files.bucket).expand(
+            file=XComArg(files)
+        )
+        total(lines=counts)
+
+What data types can be expanded?
+================================
+
+Currently it is only possible to map against a dict, a list, or one of those types stored in XCom as the result of a task.
+
+If an upstream task returns an unmappable type, the mapped task will fail at run-time with an ``UnmappableXComTypePushed`` exception. For instance, you can't have the upstream task return a plain string – it must be a list or a dict.
+
+Placing limits on mapped tasks
+==============================
+
+There are two limits that you can place on a task: how many mapped task instances can be created as the result of expansion, and how many of the mapped task can run at once
```

Comment:
> how many of the mapped task can run at once

some ambiguity here

you refer to "mapped task" both as the "parent" task (elsewhere) and it seems as the actual generated task instances (here).  may be possible to clarify here you're talking about total mapped instances and not just num "parent" mapped tasks

Reply:
parent task should be the upstream task/source of the mapping list. Can you remember where I referred to it like that?

Reply:
when i say "parent" i am referring to the mapped operator instance essentially.... which is itself never gonna be a task ... the tasks that get created are the instances that operato on each input.  and that's what you mean by `how many of the mapped task can run at once` no? like the instances -- not the "parent".  as to where, there are a lot of places where it's a little ambiguous (though not necessarily insufficiently clear).  e.g. L24 and L26.  L24 says "before a mapped task is executed the scheduler will create N copies of the task" so here you are referring to mapped task as "parent".   and same on L26 it is mapped task as parent: "collective output of a mapped task" meaning the collection of mapped _TIs_.  but when you refer "mapped task" in the line referenced above (L228) you are referring to the TIs and that's the ambiguity i'm talking about.  not big deal just a suggestion.